diff --git a/lib/Model/AbstractShip.php b/lib/Model/AbstractShip.php
index 6532e95..83b4647 100644
--- a/lib/Model/AbstractShip.php
+++ b/lib/Model/AbstractShip.php
@@ -104,9 +104,4 @@ class AbstractShip
     {
         $this->id = $id;
     }
-
-    public function getType()
-    {
-        return 'Empire';
-    }
 }
diff --git a/lib/Model/Ship.php b/lib/Model/Ship.php
index 8d89a01..b054138 100644
--- a/lib/Model/Ship.php
+++ b/lib/Model/Ship.php
@@ -34,4 +34,9 @@ class Ship extends AbstractShip
     {
         return !$this->underRepair;
     }
+
+    public function getType()
+    {
+        return 'Empire';
+    }
 }
