diff --git a/lib/Model/AbstractShip.php b/lib/Model/AbstractShip.php
index f4f8a46..e796b1d 100644
--- a/lib/Model/AbstractShip.php
+++ b/lib/Model/AbstractShip.php
@@ -8,8 +8,6 @@ class AbstractShip
 
     private $weaponPower = 0;
 
-    private $jediFactor = 0;
-
     private $strength = 0;
 
     private $underRepair;
@@ -57,7 +55,7 @@ class AbstractShip
                 '%s: %s/%s/%s',
                 $this->name,
                 $this->weaponPower,
-                $this->jediFactor,
+                $this->getJediFactor(),
                 $this->strength
             );
         } else {
@@ -65,7 +63,7 @@ class AbstractShip
                 '%s: w:%s, j:%s, s:%s',
                 $this->name,
                 $this->weaponPower,
-                $this->jediFactor,
+                $this->getJediFactor(),
                 $this->strength
             );
         }
@@ -85,14 +83,6 @@ class AbstractShip
     }
 
     /**
-     * @return int
-     */
-    public function getJediFactor()
-    {
-        return $this->jediFactor;
-    }
-
-    /**
      * @param string $name
      */
     public function setName($name)
@@ -109,14 +99,6 @@ class AbstractShip
     }
 
     /**
-     * @param int $jediFactor
-     */
-    public function setJediFactor($jediFactor)
-    {
-        $this->jediFactor = $jediFactor;
-    }
-
-    /**
      * @return int
      */
     public function getId()
diff --git a/lib/Model/Ship.php b/lib/Model/Ship.php
index 8e0d822..1432749 100644
--- a/lib/Model/Ship.php
+++ b/lib/Model/Ship.php
@@ -2,4 +2,21 @@
 
 class Ship extends AbstractShip
 {
+    private $jediFactor = 0;
+
+    /**
+     * @return int
+     */
+    public function getJediFactor()
+    {
+        return $this->jediFactor;
+    }
+
+    /**
+     * @param int $jediFactor
+     */
+    public function setJediFactor($jediFactor)
+    {
+        $this->jediFactor = $jediFactor;
+    }
 }
diff --git a/lib/Service/ShipLoader.php b/lib/Service/ShipLoader.php
index 01481fe..b01abbc 100644
--- a/lib/Service/ShipLoader.php
+++ b/lib/Service/ShipLoader.php
@@ -48,11 +48,11 @@ class ShipLoader
             $ship = new RebelShip($shipData['name']);
         } else {
             $ship = new Ship($shipData['name']);
+            $ship->setJediFactor($shipData['jedi_factor']);
         }
 
         $ship->setId($shipData['id']);
         $ship->setWeaponPower($shipData['weapon_power']);
-        $ship->setJediFactor($shipData['jedi_factor']);
         $ship->setStrength($shipData['strength']);
 
         return $ship;
