diff --git a/lib/Model/AbstractShip.php b/lib/Model/AbstractShip.php
index e796b1d..6532e95 100644
--- a/lib/Model/AbstractShip.php
+++ b/lib/Model/AbstractShip.php
@@ -10,18 +10,9 @@ class AbstractShip
 
     private $strength = 0;
 
-    private $underRepair;
-
     public function __construct($name)
     {
         $this->name = $name;
-        // randomly put this ship under repair
-        $this->underRepair = mt_rand(1, 100) < 30;
-    }
-
-    public function isFunctional()
-    {
-        return !$this->underRepair;
     }
 
     public function sayHello()
diff --git a/lib/Model/Ship.php b/lib/Model/Ship.php
index 1432749..8d89a01 100644
--- a/lib/Model/Ship.php
+++ b/lib/Model/Ship.php
@@ -4,6 +4,16 @@ class Ship extends AbstractShip
 {
     private $jediFactor = 0;
 
+    private $underRepair;
+
+    public function __construct($name)
+    {
+        parent::__construct($name);
+
+        // randomly put this ship under repair
+        $this->underRepair = mt_rand(1, 100) < 30;
+    }
+
     /**
      * @return int
      */
@@ -19,4 +29,9 @@ class Ship extends AbstractShip
     {
         $this->jediFactor = $jediFactor;
     }
+
+    public function isFunctional()
+    {
+        return !$this->underRepair;
+    }
 }
