diff --git a/battle.php b/battle.php
index 98e643a..031c777 100644
--- a/battle.php
+++ b/battle.php
@@ -84,6 +84,7 @@ $battleResult = $battleManager->battle($ship1, $ship1Quantity, $ship2, $ship2Qua
                         <?php else: ?>
                             overpowered and destroyed the <?php echo $battleResult->getLosingShip()->getName() ?>s
                         <?php endif; ?>
+                        (remaining strength: <?php echo $battleResult->getWinningShip()->getStrength(); ?>)
                     <?php endif; ?>
                 </p>
             </div>
diff --git a/lib/BattleManager.php b/lib/BattleManager.php
index 2d53460..e4dc5bc 100644
--- a/lib/BattleManager.php
+++ b/lib/BattleManager.php
@@ -34,6 +34,10 @@ class BattleManager
             $ship2Health = $ship2Health - ($ship1->getWeaponPower() * $ship1Quantity);
         }
 
+        // update the strengths on the ships, so we can show this
+        $ship1->setStrength($ship1Health);
+        $ship2->setStrength($ship2Health);
+
         if ($ship1Health <= 0 && $ship2Health <= 0) {
             // they destroyed each other
             $winningShip = null;
