diff --git a/lib/BattleResult.php b/lib/BattleResult.php
index 35a6178..2603785 100644
--- a/lib/BattleResult.php
+++ b/lib/BattleResult.php
@@ -6,10 +6,39 @@ class BattleResult
     private $winningShip;
     private $losingShip;
 
-    public function __construct($usedJediPowers, $winningShip, $losingShip)
+    /**
+     * @param Ship $winningShip
+     * @param Ship $losingShip
+     * @param boolean $usedJediPowers
+     */
+    public function __construct($usedJediPowers, Ship $winningShip, Ship $losingShip)
     {
         $this->usedJediPowers = $usedJediPowers;
         $this->winningShip = $winningShip;
         $this->losingShip = $losingShip;
     }
+
+    /**
+     * @return boolean
+     */
+    public function isUsedJediPowers()
+    {
+        return $this->usedJediPowers;
+    }
+
+    /**
+     * @return Ship
+     */
+    public function getWinningShip()
+    {
+        return $this->winningShip;
+    }
+
+    /**
+     * @return Ship
+     */
+    public function getLosingShip()
+    {
+        return $this->losingShip;
+    }
 }
