diff --git a/bootstrap.php b/bootstrap.php
index 872af88..063bfaa 100644
--- a/bootstrap.php
+++ b/bootstrap.php
@@ -3,3 +3,4 @@
 require_once __DIR__.'/lib/Ship.php';
 require_once __DIR__.'/lib/BattleManager.php';
 require_once __DIR__.'/lib/ShipLoader.php';
+require_once __DIR__.'/lib/BattleResult.php';
diff --git a/lib/BattleManager.php b/lib/BattleManager.php
index cb2e232..188d245 100644
--- a/lib/BattleManager.php
+++ b/lib/BattleManager.php
@@ -49,11 +49,7 @@ class BattleManager
             $usedJediPowers = $ship1UsedJediPowers;
         }
 
-        return array(
-            'winning_ship' => $winningShip,
-            'losing_ship' => $losingShip,
-            'used_jedi_powers' => $usedJediPowers,
-        );
+        return new BattleResult($usedJediPowers, $winningShip, $losingShip);
     }
 
     private function didJediDestroyShipUsingTheForce(Ship $ship)
diff --git a/lib/BattleResult.php b/lib/BattleResult.php
new file mode 100644
index 0000000..35a6178
--- /dev/null
+++ b/lib/BattleResult.php
@@ -0,0 +1,15 @@
+<?php
+
+class BattleResult
+{
+    private $usedJediPowers;
+    private $winningShip;
+    private $losingShip;
+
+    public function __construct($usedJediPowers, $winningShip, $losingShip)
+    {
+        $this->usedJediPowers = $usedJediPowers;
+        $this->winningShip = $winningShip;
+        $this->losingShip = $losingShip;
+    }
+}
