diff --git a/lib/Service/LoggableShipStorage.php b/lib/Service/LoggableShipStorage.php
index c31b46f..6ba2376 100644
--- a/lib/Service/LoggableShipStorage.php
+++ b/lib/Service/LoggableShipStorage.php
@@ -13,11 +13,21 @@ class LoggableShipStorage implements ShipStorageInterface
 
     public function fetchAllShipsData()
     {
-        return $this->shipStorage->fetchAllShipsData();
+        $ships =  $this->shipStorage->fetchAllShipsData();
+
+        $this->log(sprintf('Just fetched %s ships', count($ships)));
+
+        return $ships;
     }
 
     public function fetchSingleShipData($id)
     {
         return $this->shipStorage->fetchSingleShipData($id);
     }
+
+    private function log($message)
+    {
+        // todo - actually log this somewhere, instead of printing!
+        echo $message;
+    }
 }
