diff --git a/lib/Service/BattleManager.php b/lib/Service/BattleManager.php
index 1899c86..8f2cf7d 100644
--- a/lib/Service/BattleManager.php
+++ b/lib/Service/BattleManager.php
@@ -2,8 +2,11 @@
 
 class BattleManager
 {
+    // normal battle mode
     const TYPE_NORMAL = 'normal';
+    // don't allow jedi powers
     const TYPE_NO_JEDI = 'no_jedi';
+    // you can *only* win with jedi powers
     const TYPE_ONLY_JEDI = 'only_jedi';
 
     /**
