diff --git a/battle.php b/battle.php
index e181620..f2b949b 100644
--- a/battle.php
+++ b/battle.php
@@ -1,12 +1,8 @@
 <?php
 require __DIR__.'/bootstrap.php';
 
-$pdo = new PDO(
-    $configuration['db_dsn'],
-    $configuration['db_user'],
-    $configuration['db_pass']
-);
-$pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+$container = new Container();
+$pdo = $container->getPDO();
 
 $shipLoader = new ShipLoader($pdo);
 
diff --git a/bootstrap.php b/bootstrap.php
index c07d335..769dfd7 100644
--- a/bootstrap.php
+++ b/bootstrap.php
@@ -6,6 +6,7 @@ $configuration = array(
     'db_pass' => null,
 );
 
+require_once __DIR__.'/lib/Container.php';
 require_once __DIR__.'/lib/Ship.php';
 require_once __DIR__.'/lib/BattleManager.php';
 require_once __DIR__.'/lib/ShipLoader.php';
diff --git a/index.php b/index.php
index 3adcaee..5590309 100644
--- a/index.php
+++ b/index.php
@@ -1,12 +1,8 @@
 <?php
 require __DIR__.'/bootstrap.php';
 
-$pdo = new PDO(
-    $configuration['db_dsn'],
-    $configuration['db_user'],
-    $configuration['db_pass']
-);
-$pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+$container = new Container();
+$pdo = $container->getPDO();
 
 $shipLoader = new ShipLoader($pdo);
 $ships = $shipLoader->getShips();
diff --git a/lib/Container.php b/lib/Container.php
new file mode 100644
index 0000000..a8039c0
--- /dev/null
+++ b/lib/Container.php
@@ -0,0 +1,25 @@
+<?php
+
+class Container
+{
+    /**
+     * @return PDO
+     */
+    public function getPDO()
+    {
+        $configuration = array(
+            'db_dsn'  => 'mysql:host=localhost;dbname=oo_battle',
+            'db_user' => 'root',
+            'db_pass' => null,
+        );
+
+        $pdo = new PDO(
+            $configuration['db_dsn'],
+            $configuration['db_user'],
+            $configuration['db_pass']
+        );
+        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+
+        return $pdo;
+    }
+}
