diff --git a/lib/Container.php b/lib/Container.php
index b4ee023..b096e8a 100644
--- a/lib/Container.php
+++ b/lib/Container.php
@@ -4,6 +4,8 @@ class Container
 {
     private $configuration;
 
+    private $pdo;
+
     public function __construct(array $configuration)
     {
         $this->configuration = $configuration;
@@ -14,13 +16,16 @@ class Container
      */
     public function getPDO()
     {
-        $pdo = new PDO(
-            $this->configuration['db_dsn'],
-            $this->configuration['db_user'],
-            $this->configuration['db_pass']
-        );
-        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        if ($this->pdo === null) {
+            $this->pdo = new PDO(
+                $this->configuration['db_dsn'],
+                $this->configuration['db_user'],
+                $this->configuration['db_pass']
+            );
+
+            $this->pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        }
 
-        return $pdo;
+        return $this->pdo;
     }
 }
