diff --git a/bootstrap.php b/bootstrap.php
index 769dfd7..6b701be 100644
--- a/bootstrap.php
+++ b/bootstrap.php
@@ -6,8 +6,8 @@ $configuration = array(
     'db_pass' => null,
 );
 
-require_once __DIR__.'/lib/Container.php';
-require_once __DIR__.'/lib/Ship.php';
-require_once __DIR__.'/lib/BattleManager.php';
-require_once __DIR__.'/lib/ShipLoader.php';
-require_once __DIR__.'/lib/BattleResult.php';
+require_once __DIR__.'/lib/Service/Container.php';
+require_once __DIR__.'/lib/Model/Ship.php';
+require_once __DIR__.'/lib/Service/BattleManager.php';
+require_once __DIR__.'/lib/Service/ShipLoader.php';
+require_once __DIR__.'/lib/Model/BattleResult.php';
diff --git a/lib/BattleResult.php b/lib/Model/BattleResult.php
similarity index 100%
rename from lib/BattleResult.php
rename to lib/Model/BattleResult.php
diff --git a/lib/Ship.php b/lib/Model/Ship.php
similarity index 100%
rename from lib/Ship.php
rename to lib/Model/Ship.php
diff --git a/lib/BattleManager.php b/lib/Service/BattleManager.php
similarity index 100%
rename from lib/BattleManager.php
rename to lib/Service/BattleManager.php
diff --git a/lib/Container.php b/lib/Service/Container.php
similarity index 100%
rename from lib/Container.php
rename to lib/Service/Container.php
diff --git a/lib/ShipLoader.php b/lib/Service/ShipLoader.php
similarity index 100%
rename from lib/ShipLoader.php
rename to lib/Service/ShipLoader.php
