diff --git a/battle.php b/battle.php
index db37a2e..a2283bc 100644
--- a/battle.php
+++ b/battle.php
@@ -1,7 +1,7 @@
 <?php
 require __DIR__.'/bootstrap.php';
 
-$container = new Container();
+$container = new Container($configuration);
 $pdo = $container->getPDO();
 
 $shipLoader = new ShipLoader($pdo);
diff --git a/index.php b/index.php
index 5590309..57e12ef 100644
--- a/index.php
+++ b/index.php
@@ -1,7 +1,7 @@
 <?php
 require __DIR__.'/bootstrap.php';
 
-$container = new Container();
+$container = new Container($configuration);
 $pdo = $container->getPDO();
 
 $shipLoader = new ShipLoader($pdo);
diff --git a/lib/Container.php b/lib/Container.php
index a8039c0..e54559c 100644
--- a/lib/Container.php
+++ b/lib/Container.php
@@ -2,6 +2,13 @@
 
 class Container
 {
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
     /**
      * @return PDO
      */
