diff --git a/battle.php b/battle.php
index f2b949b..1b9607e 100644
--- a/battle.php
+++ b/battle.php
@@ -1,7 +1,7 @@
 <?php
 require __DIR__.'/bootstrap.php';
 
-$container = new Container();
+$container = new Container($configuration);
 $pdo = $container->getPDO();
 
 $shipLoader = new ShipLoader($pdo);
diff --git a/index.php b/index.php
index 5590309..57e12ef 100644
--- a/index.php
+++ b/index.php
@@ -1,7 +1,7 @@
 <?php
 require __DIR__.'/bootstrap.php';
 
-$container = new Container();
+$container = new Container($configuration);
 $pdo = $container->getPDO();
 
 $shipLoader = new ShipLoader($pdo);
diff --git a/lib/Container.php b/lib/Container.php
index a8039c0..b4ee023 100644
--- a/lib/Container.php
+++ b/lib/Container.php
@@ -2,21 +2,22 @@
 
 class Container
 {
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
     /**
      * @return PDO
      */
     public function getPDO()
     {
-        $configuration = array(
-            'db_dsn'  => 'mysql:host=localhost;dbname=oo_battle',
-            'db_user' => 'root',
-            'db_pass' => null,
-        );
-
         $pdo = new PDO(
-            $configuration['db_dsn'],
-            $configuration['db_user'],
-            $configuration['db_pass']
+            $this->configuration['db_dsn'],
+            $this->configuration['db_user'],
+            $this->configuration['db_pass']
         );
         $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
 
