diff --git a/index.php b/index.php
index 86fc0ed..fc32144 100644
--- a/index.php
+++ b/index.php
@@ -88,11 +88,11 @@ if (isset($_GET['error'])) {
                     <form method="POST" action="/battle.php">
                         <h2 class="text-center">The Mission</h2>
                         <input class="center-block form-control text-field" type="text" name="ship1_quantity" placeholder="Enter Number of Ships" />
-                        <select class="center-block form-control btn drp-dwn-width btn-default btn-lg dropdown-toggle" name="ship1_name">
+                        <select class="center-block form-control btn drp-dwn-width btn-default btn-lg dropdown-toggle" name="ship1_id">
                             <option value="">Choose a Ship</option>
-                            <?php foreach ($ships as $key => $ship): ?>
+                            <?php foreach ($ships as $ship): ?>
                                 <?php if ($ship->isFunctional()): ?>
-                                    <option value="<?php echo $key; ?>"><?php echo $ship->getNameAndSpecs(); ?></option>
+                                    <option value="<?php echo $ship->getId(); ?>"><?php echo $ship->getNameAndSpecs(); ?></option>
                                 <?php endif; ?>
                             <?php endforeach; ?>
                         </select>
@@ -100,11 +100,11 @@ if (isset($_GET['error'])) {
                         <p class="text-center">AGAINST</p>
                         <br>
                         <input class="center-block form-control text-field" type="text" name="ship2_quantity" placeholder="Enter Number of Ships" />
-                        <select class="center-block form-control btn drp-dwn-width btn-default btn-lg dropdown-toggle" name="ship2_name">
+                        <select class="center-block form-control btn drp-dwn-width btn-default btn-lg dropdown-toggle" name="ship2_id">
                             <option value="">Choose a Ship</option>
-                            <?php foreach ($ships as $key => $ship): ?>
+                            <?php foreach ($ships as $ship): ?>
                                 <?php if ($ship->isFunctional()): ?>
-                                    <option value="<?php echo $key; ?>"><?php echo $ship->getNameAndSpecs(); ?></option>
+                                    <option value="<?php echo $ship->getId(); ?>"><?php echo $ship->getNameAndSpecs(); ?></option>
                                 <?php endif; ?>
                             <?php endforeach; ?>
                         </select>
diff --git a/lib/Ship.php b/lib/Ship.php
index 76b164e..80c1744 100644
--- a/lib/Ship.php
+++ b/lib/Ship.php
@@ -2,6 +2,8 @@
 
 class Ship
 {
+    private $id;
+
     private $name;
 
     private $weaponPower = 0;
@@ -113,4 +115,20 @@ class Ship
     {
         $this->jediFactor = $jediFactor;
     }
+
+    /**
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     */
+    public function setId($id)
+    {
+        $this->id = $id;
+    }
 }
diff --git a/lib/ShipLoader.php b/lib/ShipLoader.php
index 81b2f7d..ad0cb0a 100644
--- a/lib/ShipLoader.php
+++ b/lib/ShipLoader.php
@@ -10,6 +10,7 @@ class ShipLoader
 
         foreach ($shipsData as $shipData) {
             $ship = new Ship($shipData['name']);
+            $ship->setId($shipData['id']);
             $ship->setWeaponPower($shipData['weapon_power']);
             $ship->setJediFactor($shipData['jedi_factor']);
             $ship->setStrength($shipData['strength']);
