diff --git a/lib/ShipLoader.php b/lib/ShipLoader.php
index ca39c7c..d018e0b 100644
--- a/lib/ShipLoader.php
+++ b/lib/ShipLoader.php
@@ -8,6 +8,12 @@ class ShipLoader
      */
     public function getShips()
     {
+        $pdo = new PDO('mysql:host=localhost;dbname=oo_battle', 'root');
+        $statement = $pdo->prepare('SELECT * FROM ship');
+        $statement->execute();
+        $shipsArray = $statement->fetchAll(PDO::FETCH_ASSOC);
+        var_dump($shipsArray);die;
+
         $ships = array();
 
         $ship = new Ship('Jedi Starfighter');
