diff --git a/lib/ShipLoader.php b/lib/ShipLoader.php
index f888043..8c6d815 100644
--- a/lib/ShipLoader.php
+++ b/lib/ShipLoader.php
@@ -24,8 +24,7 @@ class ShipLoader
      */
     public function findOneById($id)
     {
-        $pdo = new PDO('mysql:host=localhost;dbname=oo_battle', 'root');
-        $statement = $pdo->prepare('SELECT * FROM ship WHERE id = :id');
+        $statement = $this->getPDO()->prepare('SELECT * FROM ship WHERE id = :id');
         $statement->execute(array('id' => $id));
         $shipArray = $statement->fetch(PDO::FETCH_ASSOC);
 
@@ -47,11 +46,19 @@ class ShipLoader
         return $ship;
     }
 
+    /**
+     * @return PDO
+     */
+    private function getPDO()
+    {
+        $pdo = new PDO('mysql:host=localhost;dbname=oo_battle', 'root');
+
+        return $pdo;
+    }
 
     private function queryForShips()
     {
-        $pdo = new PDO('mysql:host=localhost;dbname=oo_battle', 'root');
-        $statement = $pdo->prepare('SELECT * FROM ship');
+        $statement = $this->getPDO()->prepare('SELECT * FROM ship');
         $statement->execute();
         $shipsArray = $statement->fetchAll(PDO::FETCH_ASSOC);
 
