diff --git a/lib/Model/RebelShip.php b/lib/Model/RebelShip.php
index cf9989f..a489cd8 100644
--- a/lib/Model/RebelShip.php
+++ b/lib/Model/RebelShip.php
@@ -14,4 +14,9 @@ class RebelShip extends Ship
     {
         return 'Rebel';
     }
+
+    public function isFunctional()
+    {
+        return true;
+    }
 }
diff --git a/lib/Model/Ship.php b/lib/Model/Ship.php
index f5fe019..fb02eb2 100644
--- a/lib/Model/Ship.php
+++ b/lib/Model/Ship.php
@@ -134,7 +134,6 @@ class Ship
 
     public function getType()
     {
-        echo 'Parent function!';
         return 'Empire';
     }
 }
