diff --git a/lib/Model/RebelShip.php b/lib/Model/RebelShip.php
index 12ebd9e..32c638b 100644
--- a/lib/Model/RebelShip.php
+++ b/lib/Model/RebelShip.php
@@ -22,8 +22,6 @@ class RebelShip extends Ship
 
     public function getNameAndSpecs($useShortFormat = false)
     {
-        return $this->getSecretDoorCodeToTheDeathstar();
-
         if ($useShortFormat) {
             return sprintf(
                 '%s: %s/%s/%s (Rebel)',
diff --git a/lib/Model/Ship.php b/lib/Model/Ship.php
index 0c77749..fb02eb2 100644
--- a/lib/Model/Ship.php
+++ b/lib/Model/Ship.php
@@ -136,9 +136,4 @@ class Ship
     {
         return 'Empire';
     }
-
-    protected function getSecretDoorCodeToTheDeathstar()
-    {
-        return 'Ra1nb0ws';
-    }
 }
