diff --git a/lib/Model/RebelShip.php b/lib/Model/RebelShip.php
index 32c638b..12ebd9e 100644
--- a/lib/Model/RebelShip.php
+++ b/lib/Model/RebelShip.php
@@ -22,6 +22,8 @@ class RebelShip extends Ship
 
     public function getNameAndSpecs($useShortFormat = false)
     {
+        return $this->getSecretDoorCodeToTheDeathstar();
+
         if ($useShortFormat) {
             return sprintf(
                 '%s: %s/%s/%s (Rebel)',
diff --git a/lib/Model/Ship.php b/lib/Model/Ship.php
index fb02eb2..0437137 100644
--- a/lib/Model/Ship.php
+++ b/lib/Model/Ship.php
@@ -136,4 +136,9 @@ class Ship
     {
         return 'Empire';
     }
+
+    private function getSecretDoorCodeToTheDeathstar()
+    {
+        return 'Ra1nb0ws';
+    }
 }
