diff --git a/battle.php b/battle.php
index 88c3056..cd0f441 100644
--- a/battle.php
+++ b/battle.php
@@ -95,7 +95,7 @@ $battleResult = $battleManager->battle($ship1, $ship1Quantity, $ship2, $ship2Qua
                 <h3>Remaining Strength</h3>
                 <dl class="dl-horizontal">
                     <dt><?php echo $ship1->getName(); ?></dt>
-                    <dd><?php echo $ship1->getStrength(); ?></dd>
+                    <dd><?php echo $ship1->strength; ?></dd>
                     <dt><?php echo $ship2->getName(); ?></dt>
                     <dd><?php echo $ship2->getStrength(); ?></dd>
                 </dl>
