diff --git a/play.php b/play.php
index 56d081a..20ece57 100644
--- a/play.php
+++ b/play.php
@@ -19,6 +19,17 @@ class Ship
     {
         return $this->name;
     }
+
+    public function getNameAndSpecs()
+    {
+        return sprintf(
+            '%s (w:%s, j:%s, s:%s)',
+            $this->name,
+            $this->weaponPower,
+            $this->jediFactor,
+            $this->strength
+        );
+    }
 }
 
 // but it doesn't do anything yet...
@@ -29,6 +40,6 @@ $myShip->strength = 40;
 
 echo 'Ship Name: '.$myShip->getName();
 echo '<hr/>';
-echo 'Ship Weapon Power: '.$myShip->weaponPower;
+echo 'Ship Description: '.$myShip->getNameAndSpecs();
 echo '<hr/>';
 $myShip->sayHello();
