diff --git a/play.php b/play.php
index 20ece57..8cc57a9 100644
--- a/play.php
+++ b/play.php
@@ -32,14 +32,20 @@ class Ship
     }
 }
 
+function printShipSummary($someShip)
+{
+    echo 'Ship Name: '.$someShip->getName();
+    echo '<hr/>';
+    echo 'Ship Description: '.$someShip->getNameAndSpecs();
+    echo '<hr/>';
+    $someShip->sayHello();
+}
+
+
 // but it doesn't do anything yet...
 $myShip = new Ship();
 $myShip->name = 'TIE Fighter';
 $myShip->weaponPower = 15;
 $myShip->strength = 40;
 
-echo 'Ship Name: '.$myShip->getName();
-echo '<hr/>';
-echo 'Ship Description: '.$myShip->getNameAndSpecs();
-echo '<hr/>';
-$myShip->sayHello();
+printShipSummary($myShip);
