diff --git a/play.php b/play.php
index 2c2292d..42849f2 100644
--- a/play.php
+++ b/play.php
@@ -30,6 +30,12 @@ class Ship
             $this->strength
         );
     }
+
+    public function doesGivenShipHaveMoreStrength($givenShip)
+    {
+        return $givenShip->strength > $this->strength;
+    }
+
 }
 
 /**
@@ -59,3 +65,10 @@ $otherShip->strength = 50;
 
 echo '<hr/>';
 printShipSummary($otherShip);
+echo '<hr/>';
+
+if ($myShip->doesGivenShipHaveMoreStrength($otherShip)) {
+    echo $otherShip->name.' has more strength';
+} else {
+    echo $myShip->name.' has more strength';
+}
