diff --git a/functions.php b/functions.php
index ee4dcfc..4ddf002 100644
--- a/functions.php
+++ b/functions.php
@@ -10,28 +10,28 @@ function get_ships()
     $ship1->name = 'Jedi Starfighter';
     $ship1->weaponPower = 5;
     $ship1->jediFactor = 15;
-    $ship1->strength = 'Jar Jar Binks';
+    $ship1->setStrength('Jar Jar Binks');
     $ships['starfighter'] = $ship1;
 
     $ship2 = new Ship();
     $ship2->name = 'CloakShape Fighter';
     $ship2->weaponPower = 2;
     $ship2->jediFactor = 2;
-    $ship2->strength = 70;
+    $ship2->setStrength(70);
     $ships['cloakshape_fighter'] = $ship2;
 
     $ship3 = new Ship();
     $ship3->name = 'Super Star Destroyer';
     $ship3->weaponPower = 70;
     $ship3->jediFactor = 0;
-    $ship3->strength = 500;
+    $ship3->setStrength(500);
     $ships['super_star_destroyer'] = $ship3;
 
     $ship4 = new Ship();
     $ship4->name = 'RZ-1 A-wing interceptor';
     $ship4->weaponPower = 4;
     $ship4->jediFactor = 4;
-    $ship4->strength = 50;
+    $ship4->setStrength(50);
     $ships['rz1_a_wing_interceptor'] = $ship4;
 
     return $ships;
diff --git a/index.php b/index.php
index 3ee610c..af5c34e 100644
--- a/index.php
+++ b/index.php
@@ -68,7 +68,7 @@ if (isset($_GET['error'])) {
                             <td><?php echo $ship->getName(); ?></td>
                             <td><?php echo $ship->weaponPower; ?></td>
                             <td><?php echo $ship->jediFactor; ?></td>
-                            <td><?php echo $ship->strength; ?></td>
+                            <td><?php echo $ship->getStrength(); ?></td>
                         </tr>
                     <?php endforeach; ?>
                 </tbody>
diff --git a/lib/Ship.php b/lib/Ship.php
index e9319a3..0087f7c 100644
--- a/lib/Ship.php
+++ b/lib/Ship.php
@@ -20,6 +20,16 @@ class Ship
         return $this->name;
     }
 
+    public function setStrength($number)
+    {
+        $this->strength = $number;
+    }
+
+    public function getStrength()
+    {
+        return $this->strength;
+    }
+
     public function getNameAndSpecs()
     {
         return sprintf(
